debug/unittest/ut1/main.cpp

```diff
@@ -0,0 +1,265 @@
+   double _w;
+
+public:
+   HarmonicOscillator1D1P(const double w, WaveFunction * wf):
+      Hamiltonian(1 /*num space dimensions*/, 1 /*num particles*/, wf) {_w=w;}
+
+   // potential energy
+   double localPotentialEnergy(const double *r)
+   {
+      return (0.5*_w*_w*(*r)*(*r));
+   }
+};
+
+
+
+/*
+Trial Wave Function for 1 particle in 1 dimension, that uses two variational parameters: a and b.
+   Psi  =  exp( -b * (x-a)^2 )
+Notice that the corresponding probability density (sampling function) is Psi^2.
+*/
+class QuadrExponential1D1POrbital: public WaveFunction{
+   protected:
+      double _a, _b;
+
+   public:
+      QuadrExponential1D1POrbital(const double a, const double b):
+         WaveFunction(1 /*num space dimensions*/, 1 /*num particles*/, 1 /*num wf components*/, 2 /*num variational parameters*/) {_a=a; _b=b;}
+
+      void setVP(const double *in){
+         _a=in[0];
+         _b=in[1];
+      }
+
+      void getVP(double *out){
+         out[0]=_a; out[1]=_b;
+      }
+
+      void samplingFunction(const double *x, double *out){
+         /*
+         Compute the sampling function proto value, used in getAcceptance()
+         */
+         *out = -2.*(_b*(x[0]-_a)*(x[0]-_a));
+      }
+
+      double getAcceptance(){
+         /*
+         Compute the acceptance probability
+         */
+         return exp(getProtoNew(0)-getProtoOld(0));
+      }
+
+      double d1(const int &i, const double *x){
+         /*
+         Compute:    d/dx_i log(Psi(x))
+         */
+         return (-2.*_b*(x[0]-_a) );
+      }
+
+      double d2(const int &i, const int &j, const double *x){
+         /*
+         Compute:    d^2/dx_i^2 log(Psi(x))
+         */
+         return ( -2.*_b + (-2.*_b*(x[0]-_a))*(-2.*_b*(x[0]-_a)) ) ;
+      }
+
+      double vd1(const int &i, const double *x){
+         /*
+         Compute:    d/dalpha_i log(Psi(x))
+         where alpha are the variational parameters, in our case an array of dimension 1: alpha = (b)
+         */
+         if (i==0){
+            return (2.*_b*(x[0]-_a));
+         } else if (i==1){
+            return (-(x[0]-_a)*(x[0]-_a));
+         } else{
+            using namespace std;
+            cout << "ERRORE vd1 QuadrExponential! " << endl;
+            return 0.;
+         }
+      }
+};
+
+
+
+int main(){
+    using namespace std;
+
+    // parameters
+    const long Nmc = 100000l;
+    const double TINY = 0.000001;
+
+    // variational parameters
+    double a = 0.37;
+    double sqrtb = 1.18;
+    double b = sqrtb * sqrtb;
+    double p1 = - sqrtb * a;
+    double p2 = sqrtb;
+
+
+
+
+    // create the NN with the right structure and parameters
+    FeedForwardNeuralNetwork * ffnn = new FeedForwardNeuralNetwork(2, 2, 2);
+    GaussianActivationFunction * gss_actf = new GaussianActivationFunction();
+    IdentityActivationFunction * id_actf = new IdentityActivationFunction();
+    ffnn->setLayerActivationFunction(1, gss_actf);
+    ffnn->setLayerActivationFunction(2, id_actf);
```

Comment:
Just curious: Why and when should I use these? Why did you do it like that here?

Reply:
It is explained at the beginning of the file, in the comment. I did that for having a NN that actually reproduces a gaussian

Reply:
Might be useful for my fitting?!